app/lib/katello/repo_discovery.rb

```diff
@@ -42,6 +44,73 @@ def run(resume_point)
 
     private
 
+    def parse_parameter(field_value)
+      field_value.sub!(/^[ \t]*/, '')
+      i = field_value.index(/[ \t=;,]/) || field_value.length
```

Comment:
One caveat I found:

```ruby
s = "<http://pulpcore-api/v2/container-image-name/tags/list?n=100&last=last-tag-name>; rel=\"next\""

m  = s.match(/<(.*)>/)
=> "; rel=\"next\""

m.post_match.split(/[ \t=;,]/, 2).map(&:strip)
=> ["", "rel=\"next\""]



```

The non-whitespace delimiters (here a semicolon) between the URL and the key-value pairs need to be cut out.

Reply:
Actually, one amendment:

if `field_value` is `"rel=\"next\" anchor=\"blah\""`, then the anchor part will end up in `value`.

If we take out the limit of `2`, then the first and second entries into the array that `strip` returns would definitely include the name and value of the first argument.

```ruby
params = "rel=\"next\" anchor=\"blah\""
[59] pry(main)> n,v = params.split(/[ \t=;,]/).map(&:strip)
=> ["rel", "\"next\"", "anchor", "\"blah\""]
[61] pry(main)> n
=> "rel"
[62] pry(main)> v
=> "\"next\""
```

I realize the split method won't jiving perfectly with the slicing that you're doing to the args string so that you can loop through it and collect up all of the key-value pairs, but hopefully it serves as some good inspiration/learning/etc.

Reply:
I think, for a general case like `<http://example.com>; title=foo; rel="bar next baz"` we don't get away with anything simpler than tediously following the RFC 8288 algorithm.

Reply:
Alright, I'm fine with keeping it as-is to follow the spec guidelines.